eftcamb/03_abstract_EFTCAMB_cache.f90

```diff
@@ -332,10 +337,39 @@ subroutine EFTCAMBTimestepCacheInit( self )
         self%EFTAT         = 0._dl
         self%EFTBT         = 0._dl
         self%EFTDT         = 0._dl
+        ! 12) Kinetic and Gradient quantities for the stability check:
+        self%EFT_kinetic   = 0._dl
+        self%EFT_gradient  = 0._dl
 
     end subroutine EFTCAMBTimestepCacheInit
 
     ! ---------------------------------------------------------------------------------------------
+    !> Subroutine that check if an element of the EFTCAMB_timestep_cache is Nan.
+    subroutine EFTCAMBTimestepCacheIsNan( self, HaveNan )
+
+      implicit none
+
+      class(EFTCAMB_timestep_cache), intent(in)  :: self    !< The base class.
+      logical, intent(out)                       :: HaveNan !< Logical variable which describes the presence of a Nan variable.
+                                                            !< If an element of the EFTCAMB_timestep_cache is Nan, you get HaveNan=.True.
+
+      HaveNan = IsNaN(self%a).or.IsNaN(self%tau).or.IsNaN(self%k).or.IsNaN(self%adotoa).or.IsNaN(self%Hdot).or.IsNaN(self%Hdotdot).or.IsNaN(self%grhoa2)&
```

Comment:
I would reshuffle a bit the structure of this part of code to make it clearer. Something like:
HaveNaN = .False.
HaveNaN = HaveNaN .or. IsNaN(self%a)
HaveNaN = HaveNaN .or. IsNaN(self%tau)
...
This adds clarity to the code and allows to check wether a cache term is there or not